src/node_platform.cc

```diff
@@ -0,0 +1,201 @@
+#include "node_platform.h"
+
+namespace node {
+
+using v8::Isolate;
+using v8::Platform;
+using v8::Task;
+using v8::TracingController;
+
+static void FlushTasks(uv_async_t* handle) {
+  NodePlatform* platform = static_cast<NodePlatform*>(handle->data);
+  platform->FlushForegroundTasksInternal();
+}
+
+static void DeleteHandle(uv_handle_t* handle) {
+  delete handle;
+}
+
+static void BackgroundRunner(void* data) {
+  TaskQueue<Task>* background_tasks = static_cast<TaskQueue<Task>*>(data);
+  Task* task = NULL;
+  while ((task = background_tasks->BlockingPop())) {
+    task->Run();
+    delete task;
+    background_tasks->NotifyOfCompletion();
+  }
+}
+
+NodePlatform::NodePlatform(int thread_pool_size, uv_loop_t* loop,
+                           TracingController* tracing_controller)
+    : loop_(loop) {
+  uv_async_init(loop, &flush_tasks_, FlushTasks);
+  flush_tasks_.data = static_cast<void*>(this);
+  uv_unref(reinterpret_cast<uv_handle_t*>(&flush_tasks_));
+  if (tracing_controller) {
+    tracing_controller_.reset(tracing_controller);
+  } else {
+    TracingController* controller = new TracingController();
+    tracing_controller_.reset(controller);
+  }
+  threads_.resize(thread_pool_size);
+  for (int i = 0; i < thread_pool_size; i++) {
+    uv_thread_create(&threads_[i], BackgroundRunner, &background_tasks_);
```

Comment:
You should check the return value and be prepared for failure.  (Think `ulimit -u`.)